pkg/resources/machinecontroller/deployment.go

```diff
@@ -193,6 +193,10 @@ func DeploymentCreatorWithoutInitWrapper(data machinecontrollerData) reconciling
 	}
 }
 
+func isExternalCloudProvider(data machinecontrollerData) bool {
```

Comment:
This part is not needed. If you want to always deploy the CCM as external provider, I suggest that you do it over mutation webhook like here: https://github.com/kubermatic/kubermatic/pull/7543/files#diff-8bbc7dde93249c1e7069d06767c89f4476b29eaaa92d861111a5843a73d2c6daR229-R232

There is no need to change the core deployment functions.

Reply:
Done